pandas/core/construction.py

```diff
@@ -502,9 +502,7 @@ def sanitize_masked_array(data: ma.MaskedArray) -> np.ndarray:
     if mask.any():
         dtype, fill_value = maybe_promote(data.dtype, np.nan)
         dtype = cast(np.dtype, dtype)
-        # Incompatible types in assignment (expression has type "ndarray[Any,
-        # dtype[Any]]", variable has type "MaskedArray[Any, Any]")
-        data = data.astype(dtype, copy=True)  # type: ignore[assignment]
+        data = ma.asarray(data.astype(dtype, copy=True))
```

Comment:
thanks for your PR - would it work to just rename `data`? looks like the issue that it's getting reassigned to a different type

something like
```
    data_newname = data.astype(dtype, copy=True)
    data_newname.soften_mask()
    data_newname[mask] = fill_value
    return data_newname
else:
    return data.copy()
```
(the hardest part would arguably be coming up with a good name, i.e. not `data_newname` but something more descriptive, but I haven't looked at this carefully enough yet to think of what)

Reply:
Thank you, @MarcoGorelli for your comment. I will do, as you suggest. What do you think, could be `sanitized_data` or `sanitized_ma` a suitable new name for `data`?